### ISSUE TITLE
Upstream the driver

### ISSUE THREAD

**Issue by FFY00** [Henneberg-Systemdesign/cp2130#38]
Hello, what do you think about upstreaming the driver? It would be really helpful to have it in the official kernel tree.

**Comment by jh-hsd** [comment]
I agree, I have this on my list. I will try to prepare a patch next week and see what happens ...

**Comment by FFY00** [comment]
Ok, thank you :grin:.

**Comment by FFY00** [comment]
Hello, any news?

**Comment by jh-hsd** [comment]
I have sent a patch, however, it will require relevant changes to be accepted. So please be patient.

**Comment by perigoso** [comment]
Randomly stumbled onto this look through old repos, did you ever get it merged?

**Comment by jh-hsd** [comment]
No. The main reason is that they will not accept the current special sysfs setup. Kernel people would prefer something from device tree overlays and/or ACPI. However, this is not so easy since the current device tree/ACPI features do not allow a flexible USB device add/remove and I did not find time to figure out how this can work. See the whole story at:
https://www.spinics.net/lists/linux-spi/msg17314.html
Now the chip is obsolete and the driver would need quite some rework to match current kernel API better. Not sure if I will ever try to get this into the kernel since the chip is not available anymore.
